token-lending/program/src/processor.rs

```diff
@@ -1714,6 +1715,87 @@ fn unpack_mint(data: &[u8]) -> Result<Mint, LendingError> {
     Mint::unpack(data).map_err(|_| LendingError::InvalidTokenMint)
 }
 
+fn get_pyth_product_quote_currency(pyth_product: &pyth::Product) -> Result<[u8; 32], ProgramError> {
+    const LEN: usize = 14;
+    const KEY: &[u8; LEN] = b"quote_currency";
+
+    let mut offset = 0;
+    while offset < pyth::PROD_ATTR_SIZE {
+        let mut len = pyth_product.attr[offset] as usize;
+        offset += 1;
+
+        if len == LEN {
+            let key = &pyth_product.attr[offset..(offset + len)];
+            if key == KEY {
+                offset += len;
+                len = pyth_product.attr[offset] as usize;
```

Comment:
Can you add a check that `len` is less than or equal to 32?

Reply:
Done.